Api/Presentation/IncomesController.cs

```diff
@@ -1,47 +1,86 @@
 using Api.Domain;
-
+using Api.Domain.Services;
 using AutoMapper;
-
 using Microsoft.AspNetCore.Mvc;
 
 [ApiController]
 [Route("api/incomes")]
 public class IncomesController : ControllerBase
 {
     private readonly IMapper _mapper;
+    private readonly IIncomeService _incomeService;
 
-    public IncomesController(IMapper mapper)
+    public IncomesController(IMapper mapper, IIncomeService incomeService)
     {
         _mapper = mapper;
+        _incomeService = incomeService;
     }
 
     [HttpPost]
     public async Task<IActionResult> LogIncome([FromBody] CreateIncomeRequest createIncomeRequest)
     {
-        return Ok("Income logged successfully.");
+        if (createIncomeRequest == null)
+            return BadRequest("Invalid income request.");
+
+        var income = _mapper.Map<Income>(createIncomeRequest);
```

Comment:
add validations using fluent validations for all income-handling methods that needs